### PR TITLE
Implemented pprof

### PR DESCRIPTION
**Pull request by ellistarn**
**1. Issue, if available:**

**2. Description of changes:**
😮😮😮😮😮
😮😮😮😮😮
😮😮😮😮😮
karpenter-profile

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dewjam — COMMENTED** [review]
Do we want the Profiling endpoint to be enabled by default?  Or should we add a command line flag to explicitly enable Profiling similar to kube-controller-manager?

File: website/content/en/preview/development-guide.md [hunk 1]
```diff
@@ -114,3 +114,23 @@ Once you have your ECR repository provisioned, configure your Docker daemon to a
 export KO_DOCKER_REPO="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/karpenter"
 aws ecr get-login-password --region "${AWS_DEFAULT_REGION}" | docker login --username AWS --password-stdin "${KO_DOCKER_REPO}"
 ```
+
+## Profiling memory
+Karpenter exposes a pprof endpoint on its metrics port.
+
+Learn about profiling with pprof: https://jvns.ca/blog/2017/09/24/profiling-go-with-pprof/
```
Comment (chrisnegus) [inline website/content/en/preview/development-guide.md:121]:
```suggestion
Learn about profiling with [Profiling Go programs with pprof](https://jvns.ca/blog/2017/09/24/profiling-go-with-pprof/)
```

File: website/content/en/preview/development-guide.md [hunk 2]
```diff
@@ -114,3 +114,23 @@ Once you have your ECR repository provisioned, configure your Docker daemon to a
...
+
+### Prerequisites
+```
```
Comment (chrisnegus) [inline website/content/en/preview/development-guide.md:124]:
```suggestion
```bash
```

File: website/content/en/preview/development-guide.md [hunk 3]
```diff
@@ -114,3 +114,23 @@ Once you have your ECR repository provisioned, configure your Docker daemon to a
...
+brew install graphviz
+go install github.com/google/pprof@latest
+```
+
+### Get a profile
+```
```
Comment (chrisnegus) [inline website/content/en/preview/development-guide.md:130]:
```suggestion
```bash
```

**Review by chrisnegus — COMMENTED** [review]
I added a couple of small suggestions to the one doc file.

**Comment by ellistarn** [comment]
@dewjam 
> Do we want the Profiling endpoint to be enabled by default? Or should we add a command line flag to explicitly enable Profiling similar to kube-controller-manager?

I'm not savvy to the tradeoffs of running this by default. I've read about security concerns, but we already expose a broad amount of metrics on :8080 and rely on Kubernetes networking to protect the pod. Perhaps there are some performance conerns?

Seems like it might be okay according to @rakyll
https://stackoverflow.com/questions/64057727/is-it-ok-to-use-golang-pprof-on-production-without-effecting-performance

**Comment by dewjam** [comment]
@ellistarn and I spoke offline a bit more and it's probably worth enabling the debug endpoint via a flag or ENV when needed.

**Review by dewjam — APPROVED** [review]
LGTM

**Pull request merged**